### PR TITLE
Improve plot grid appearance

### LINKED ISSUE #2411: Improve plot grid appearance
I noticed some aspects of the plot that can be improved visually:

- The grid line rounding in https://github.com/emilk/egui/commit/8602326af5884b988e69dc22a090d13882d1aac2 makes the plot look a bit better when the bounds are static. However in cases where the plot bounds move very slowly, this leads to a weird effect where they are visibly jumping around between pixels.
- The origin label is drawn twice.
- Plot grid lines are transparent right now, which results in this white spot where they cross:
image

### PR DESCRIPTION
**Pull request by s-nie**
Closes #2411.

Before/After
Screenshot_20221208_144531
Screenshot_20221208_144447

### CODE REVIEW

File: crates/egui/src/widgets/plot/mod.rs [hunk 1]
```diff
@@ -331,6 +332,7 @@ impl Plot {
             show_background: true,
             show_axes: [true; 2],
             grid_spacers: [log_grid_spacer(10), log_grid_spacer(10)],
+            sharp_grid_lines: false,
```
Comment (s-nie) [inline crates/egui/src/widgets/plot/mod.rs:335]: We can discuss whether to make this the default or not. I personally don't mind the slight aliasing.

File: crates/egui/src/widgets/plot/mod.rs [hunk 2]
```diff
@@ -1290,18 +1301,30 @@ struct PreparedPlot {
...
+                    d,
+                    self.show_axes[1 - d],
+                    &mut axes_shapes,
+                    self.sharp_grid_lines,
+                );
             }
         }
 
+        // Sort the axes by contrast so that those with higher contrast are drawn in front.
+        axes_shapes.sort_by(|(_, contrast1), (_, contrast2)| contrast1.total_cmp(contrast2));
```
Comment (emilk) [inline crates/egui/src/widgets/plot/mod.rs:1324]: maybe "strength" is a slightly better word?

File: crates/egui/src/widgets/plot/mod.rs (hunk 1 above)
Comment (emilk) [reply crates/egui/src/widgets/plot/mod.rs:335]: I think I prefer it on (as is currently the case)

**Review by emilk — APPROVED** [review]
nice fixes!

File: crates/egui/src/widgets/plot/mod.rs (hunk 1 above)
Comment (s-nie) [reply crates/egui/src/widgets/plot/mod.rs]: Now enabled by default in https://github.com/emilk/egui/pull/2412/commits/848310deeb33cc9b073722cadde41e828bdea450.

File: crates/egui/src/widgets/plot/mod.rs (hunk 2 above)
Comment (s-nie) [reply crates/egui/src/widgets/plot/mod.rs]: Changed in https://github.com/emilk/egui/pull/2412/commits/848310deeb33cc9b073722cadde41e828bdea450.

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,6 +38,7 @@ NOTE: [`epaint`](crates/epaint/CHANGELOG.md), [`eframe`](crates/eframe/CHANGELOG
 * Tooltips are only shown when mouse pointer is still ([#2263](https://github.com/emilk/egui/pull/2263)).
 * Make it slightly easier to click buttons ([#2304](https://github.com/emilk/egui/pull/2304)).
 * `egui::color` has been renamed `egui::ecolor` ([#2399](https://github.com/emilk/egui/pull/2399)).
+* Improved plot grid appearance ([#2412](https://github.com/emilk/egui/pull/2412)).
```
[the remark this answers is not in the document]
Comment (s-nie) [reply CHANGELOG.md]: Moved in https://github.com/emilk/egui/pull/2412/commits/848310deeb33cc9b073722cadde41e828bdea450.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@ NOTE: [`epaint`](crates/epaint/CHANGELOG.md), [`eframe`](crates/eframe/CHANGELOG
 * MSRV (Minimum Supported Rust Version) is now `1.65.0` ([#2314](https://github.com/emilk/egui/pull/2314)).
 * ⚠️ BREAKING: egui now expects integrations to do all color blending in gamma space ([#2071](https://github.com/emilk/egui/pull/2071)).
 * ⚠️ BREAKING: if you have overlapping interactive widgets, only the top widget (last added) will be interactive ([#2244](https://github.com/emilk/egui/pull/2244)).
+* Improved plot grid appearance ([#2412](https://github.com/emilk/egui/pull/2412)).
```
Comment (emilk) [inline CHANGELOG.md:11]: This is still in the wrong place - you need to merge in `main` before moving it!

Comment (s-nie) [reply CHANGELOG.md]: Sorry about that, should be okay in https://github.com/emilk/egui/pull/2412/commits/52ee847dbc7db65ea1e9963460bf99d489308dca.

**Pull request merged**